packages/mdc-segmented-button/segmented-button/test/component.test.ts

```diff
@@ -0,0 +1,252 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {emitEvent} from '../../../../testing/dom/events';
+import {createMockFoundation} from '../../../../testing/helpers/foundation';
+// import {MDCSegmentedButtonSegmentFoundation} from '../../segment/index';
+import {MDCSegmentedButton, MDCSegmentedButtonFoundation} from '../index';
+import {strings} from '../constants';
+import {test_strings, test_indices, test_segment_ids} from './constants';
+
+const getFixtureMultiWithLabel = () => {
+  const wrapper = document.createElement('div');
+  wrapper.innerHTML = `
+    <div class="mdc-segmented-button" role="group">
+      <button class="mdc-segmented-button__segment" aria-pressed="false">
+        <div class="mdc-segmented-button__label">Segment Label</div>
+      </button>
+      <button class="mdc-segmented-button__segment" aria-pressed="false">
+        <i class="material-icons mdc-segmented-button__icon">favorite</i>
+      </button>
+      <button class="mdc-segmented-button__segment" aria-pressed="false">
+        <div class="mdc-segmented-button__label">Segment Label</div>
+        <i class="material-icons mdc-segmented-button__icon">favorite</i>
+      </button>
+    </div>
+  `;
+
+  const el = wrapper.firstElementChild as HTMLElement;
+  wrapper.removeChild(el);
+  return el;
+};
+
+class FakeSegment {
+  initialSyncWithDOM: jasmine.Spy;
+  destroy: jasmine.Spy;
+  getDefaultFoundation: jasmine.Spy;
+  setIndex: jasmine.Spy;
+  setIsSingleSelect: jasmine.Spy;
+  isSelected: jasmine.Spy;
+  setSelected: jasmine.Spy;
+  setUnselected: jasmine.Spy;
+  getSegmentId: jasmine.Spy;
+
+  constructor(_el: HTMLElement) {
+    this.initialSyncWithDOM = jasmine.createSpy('.initialSyncWithDOM');
+    this.destroy = jasmine.createSpy('.destroy');
+    this.getDefaultFoundation = jasmine.createSpy('.getDefaultFoundation');
+    this.setIndex = jasmine.createSpy('.setIndex');
+    this.setIsSingleSelect = jasmine.createSpy('.setIsSingleSelect');
+    this.isSelected = jasmine.createSpy('.isSelected').and.returnValue(false);
+    this.setSelected = jasmine.createSpy('.setSelected');
+    this.setUnselected = jasmine.createSpy('.setUnselected');
+    this.getSegmentId = jasmine.createSpy('.getSegmentId');
+  }
+}
+
+const setupTest = () => {
+  const root = getFixtureMultiWithLabel();
+  const component = new MDCSegmentedButton(root, undefined, (el: HTMLElement) => new FakeSegment(el));
+  return {root, component};
+};
+
+const setupMockFoundationTest = () => {
+  const root = getFixtureMultiWithLabel();
+  const mockFoundation = createMockFoundation(MDCSegmentedButtonFoundation);
+  const component = new MDCSegmentedButton(root, mockFoundation, (el: HTMLElement) => new FakeSegment(el));
+
+  return {root, component, mockFoundation};
+};
+
+describe('MDCSegmentedButton', () => {
```

Comment:
Can you remove foundation mocks and do real DOM testing? See Data Table component tests as an example.

Reply:
I'm mainly testing that the component's methods are just proxies to the foundation's methods. The SegmentedButton component doesn't change the DOM but calls the children Segment component methods that change the DOM. So do you want me to remove all of the mocks (like FakeSegment) to let the DOM get manipulated and then check that the DOM updated like I expected?

Reply:
By [testing behaviors](http://go/unit-test-practices?polyglot=java#behavior-testing) instead of mocking the methods, we can add new behaviors without changing existing tests. This'll allow us to refactor internal business logic without changing tests.

This is a fool-proof way of testing the component and it makes sure that it works when integrated with foundation and sub-component.

Reply:
I've removed mocks from the tests and I think I'm testing the behavior correctly, but I don't have access to the linked doc to compare.